### PR TITLE
Some small updates

### PR DESCRIPTION
**Pull request by tphalp**
This PR is a collection of necessary changes that were identified while working on the Sophtron Adapter package.

### CODE REVIEW

File: apps/server/src/test-adapter/constants.ts [hunk 1]
```diff
@@ -10,6 +10,10 @@ export const TEST_EXAMPLE_C_LABEL_TEXT = "TestExampleC Label";
 
 export const testAggregatorMemberGuid = "testAggregatorMemberGuid";
 
+export const testDataValidatorStartTimeError =
```
Comment (wesrisenmay-mx) [inline apps/server/src/test-adapter/constants.ts:13]: These are only used once and are test specific right? I'd probably hardcode them in the tests instead

Comment (tphalp) [reply apps/server/src/test-adapter/constants.ts]: I'm actually using them in the test-adapter as return values for the DataRequestValidation.transactions return value, so I'm using them in two places each.

File: apps/server/src/connect/dataEndpoints.test.ts [hunk 2]
```diff
@@ -233,29 +261,6 @@ describe("dataEndpoints", () => {
         expect(res.status).toHaveBeenCalledWith(400);
       });
 
-      it("doesn't respond with a 400 if it's TestAdapterA and there is no start or end time", async () => {
```
Comment (tphalp) [inline apps/server/src/connect/dataEndpoints.test.ts:236]: Moved down with the other validation tests.

File: apps/server/src/connect/dataEndpoints.test.ts [hunk 3]
```diff
@@ -397,6 +402,77 @@ describe("dataEndpoints", () => {
...
+        } as unknown as any;
+
+        await createTransactionsDataHandler(true)(req, res);
+
+        expect(res.send).toHaveBeenCalledWith({
+          jwt: transactionsResponse,
+        });
+      });
+
+      it("does NOT respond with a 400 if it's TestAdapterA and start_time is undefined", async () => {
```
Comment (tphalp) [inline apps/server/src/connect/dataEndpoints.test.ts:430]: This is the one moved from above.

File: apps/server/src/connect/dataEndpoints.test.ts [hunk 4]
```diff
@@ -26,9 +34,29 @@ const vcAccountsDataHandler = createAccountsDataHandler(true);
...
+  aggregator: TEST_EXAMPLE_B_AGGREGATOR_STRING,
+};
+let connectTestB: ConnectApi;
+
 describe("dataEndpoints", () => {
-  beforeEach(() => {
+  beforeEach(async () => {
     jest.restoreAllMocks();
+
+    connectTestA = new ConnectApi({
```
Comment (wesrisenmay-mx) [inline apps/server/src/connect/dataEndpoints.test.ts:51]: Why are these necessary? I see where they're passed in down below, but I don't think they're actually being used.

Comment (tphalp) [reply apps/server/src/connect/dataEndpoints.test.ts:51]: If I don't do this, then the `req.connectApi` property isn't populated, and the code doesn't work.

File: apps/server/src/connect/dataEndpoints.test.ts [hunk 5]
```diff
@@ -397,6 +402,77 @@ describe("dataEndpoints", () => {
         expect(res.send).toHaveBeenCalledWith("Something went wrong");
         expect(res.status).toHaveBeenCalledWith(400);
       });
+
+      it("returns transaction data, if it's TestAdapterB and passes the transaction validator", async () => {
```
Comment (wesrisenmay-mx) [inline apps/server/src/connect/dataEndpoints.test.ts:406]: These it statements are hard to parse. When looking at the code it should be easy for me to look through the tests and see if there's coverage for everything. We don't need to mention which aggregator it's for except for sophtron. the it statements should focus on the code that's being tested instead of the aggregator that's being tested. 

Examples: "succeeds if there is a custom validator and it passes" "succeeds if there's no custom validator" "fails if a custom validator fails" "fails if it's sophtron and it's missing a startTime" "succeeds if it's sophtron and it has a startTime" etc

File: apps/server/src/connect/dataEndpoints.test.ts [hunk 6]
```diff
@@ -27,7 +30,7 @@ const vcIdentityDataHandler = createIdentityDataHandler(true);
 const vcTransactionsDataHandler = createTransactionsDataHandler(true);
 
 describe("dataEndpoints", () => {
-  beforeEach(() => {
+  beforeEach(async () => {
```
Comment (wesrisenmay-mx) [inline apps/server/src/connect/dataEndpoints.test.ts:33]: No longer need this async

File: apps/server/cypress/e2e/suite1/testExample.cy.ts [hunk 7]
```diff
@@ -38,6 +40,96 @@ const makeABConnection = async (jobType) => {
...
+      method: "GET",
+      url: `/api${url}`,
+      failOnStatusCode: false,
+    })
+    .then((dataResponse) => {
+      expect(dataResponse.status).to.equal(400);
+    });
+};
+
+export const generateValidatorTests = ({ makeAConnection }) => {
```
Comment (wesrisenmay-mx) [inline apps/server/cypress/e2e/suite1/testExample.cy.ts:91]: The generateValidatorTests function feels like a premature abstraction. It's only used once. I'd just put the it statement directly in the describe

File: packages/utils-dev-dependency/cypress/generateDataTests.ts [hunk 8]
```diff
@@ -80,7 +80,7 @@ const verifyTransactions = ({
   shouldTestVcEndpoint,
   userId,
 }) => {
-  const url = `/data/aggregator/${aggregator}/user/${userId}/account/${accountId}/transactions${aggregator === "sophtron" ? "?start_time=2021/1/1&end_time=2024/12/31" : ""}`;
+  const url = `/data/aggregator/${aggregator}/user/${userId}/account/${accountId}/transactions${aggregator === "testExampleB" || aggregator === "sophtron" ? "?start_time=2021/1/1&end_time=2024/12/31" : ""}`;
```
Comment (wesrisenmay-mx) [inline packages/utils-dev-dependency/cypress/generateDataTests.ts:83]: We need this not care about which aggregator it is. Should we pass in an optional query string parameter?

File: apps/server/cypress/e2e/suite1/testExample.cy.ts [hunk 9]
```diff
@@ -46,5 +96,46 @@ describe("testExampleA and B aggregators", () => {
   generateDataTests({
     makeAConnection: makeABConnection,
     shouldTestVcEndpoint: true,
+    queryString: "?start_time=2021/1/1",
+  });
+
+  const jobType = JobTypes.VERIFICATION;
```
Comment (wesrisenmay-mx) [inline apps/server/cypress/e2e/suite1/testExample.cy.ts:102]: I don't think this variable needs to exist

File: apps/server/cypress/e2e/suite1/testExample.cy.ts [hunk 10]
```diff
@@ -38,6 +40,54 @@ const makeABConnection = async (jobType) => {
...
+    return accountId;
+  });
+};
+
+const verifyTransactionsValidatorSuccess = ({
+  accountId,
+  aggregator,
+  userId,
+}) => {
+  const url = `/data/aggregator/${aggregator}/user/${userId}/account/${accountId}/transactions${aggregator === TEST_EXAMPLE_B_AGGREGATOR_STRING ? "?start_time=2021/1/1" : ""}`;
```
Comment (wesrisenmay-mx) [inline apps/server/cypress/e2e/suite1/testExample.cy.ts:60]: no reason for the aggregator === TEST_EXAMPLE_B_AGGREGATOR_STRING right?

File: packages/utils-dev-dependency/cypress/generateDataTests.ts [hunk 11]
```diff
@@ -79,8 +79,16 @@ const verifyTransactions = ({
...
+  queryString = "",
+}: {
+  accountId: string;
+  aggregator: string;
+  shouldTestVcEndpoint: boolean;
+  userId: string;
+  queryString?: string;
 }) => {
-  const url = `/data/aggregator/${aggregator}/user/${userId}/account/${accountId}/transactions${aggregator === "sophtron" ? "?start_time=2021/1/1&end_time=2024/12/31" : ""}`;
+  cy.log("querystring", queryString);
```
Comment (wesrisenmay-mx) [inline packages/utils-dev-dependency/cypress/generateDataTests.ts:90]: no need to log this right?

File: apps/server/cypress/e2e/suite1/testExample.cy.ts (hunk 10 above)
Comment (tphalp) [reply apps/server/cypress/e2e/suite1/testExample.cy.ts]: Yep. Fixed.

File: apps/server/cypress/e2e/suite2/sophtron.cy.ts [hunk 12]
```diff
@@ -63,5 +63,9 @@ describe("Sophtron aggregator", () => {
     expectConnectionSuccess();
   });
 
-  generateDataTests({ makeAConnection, shouldTestVcEndpoint: true });
+  generateDataTests({
+    makeAConnection,
+    shouldTestVcEndpoint: true,
+    queryString: "?start_time=2021/1/1&end_time=2022/1/1",
```
Comment (wesrisenmay-mx) [inline apps/server/cypress/e2e/suite2/sophtron.cy.ts:69]: This variable name makes it unclear where this is going to be used. Maybe transactionsQueryString?

File: packages/utils-dev-dependency/cypress/generateDataTests.ts (hunk 11 above)
Comment (tphalp) [reply packages/utils-dev-dependency/cypress/generateDataTests.ts]: Yep, removed.

File: apps/server/cypress/e2e/suite2/sophtron.cy.ts (hunk 12 above)
Comment (tphalp) [reply apps/server/cypress/e2e/suite2/sophtron.cy.ts:69]: Renaming done

File: apps/server/cypress/e2e/suite1/testExample.cy.ts [hunk 13]
```diff
@@ -38,6 +40,55 @@ const makeABConnection = async (jobType) => {
...
+
+    return accountId;
+  });
+};
+
+const verifyTransactionsValidatorSuccess = ({
+  accountId,
+  aggregator,
+  userId,
+  queryString = "",
```
Comment (wesrisenmay-mx) [inline apps/server/cypress/e2e/suite1/testExample.cy.ts]: Why do we need an aggregator or queryString variable if it's always the same?

Comment (tphalp) [reply apps/server/cypress/e2e/suite1/testExample.cy.ts]: Good point. Removed

Comment (tphalp) [reply apps/server/cypress/e2e/suite1/testExample.cy.ts]: Removing aggregator....

Comment (tphalp) [reply apps/server/cypress/e2e/suite1/testExample.cy.ts]: Done

**Review by wesrisenmay-mx — APPROVED** [review]

**Review by emilysallstrom — APPROVED** [review]
